tensorboard/backend/event_processing/db_import_multiplexer.py

```diff
@@ -50,30 +55,39 @@ def __init__(self,
       max_reload_threads: The max number of threads that TensorBoard can use
         to reload runs. Each thread reloads one run at a time. If not provided,
         reloads runs serially (one after another).
+      use_import_op: If True, use TensorFlow's import_event() op for imports,
+        otherwise use TensorBoard's own sqlite ingestion logic.
     """
     tf.logging.info('DbImportMultiplexer initializing');
-    conn = db_connection_provider()
+    self._db_connection_provider = db_connection_provider
+    self._purge_orphaned_data = purge_orphaned_data
+    self._max_reload_threads = max_reload_threads
+    self._use_import_op = use_import_op
+    self._event_sink = None
+    self._run_loaders = {}
+
+    if self._purge_orphaned_data:
+      tf.logging.warning(
+          '--db_import does not yet support purging orphaned data')
+
+    conn = self._db_connection_provider()
     # Extract the file path of the DB from the DB connection.
-    rows = conn.execute('PRAGMA database_list;').fetchall()
-    tf.logging.info('Got rows %s', type(rows))
+    rows = conn.execute('PRAGMA database_list').fetchall()
     db_name_to_path = {row[1]: row[2] for row in rows}
     self._db_path = db_name_to_path['main']
     tf.logging.info('DbImportMultiplexer using db_path %s', self._db_path)
     # Set the DB in WAL mode so reads don't block writes.
-    # TODO(nickfelt): investigate weird errors in rollback journal mode
-    conn.execute('PRAGMA journal_mode=WAL;')
-    conn.execute('PRAGMA synchronous=NORMAL;')  # Recommended for WAL mode
-    self._run_importers = {}
-    self.purge_orphaned_data = purge_orphaned_data
-    if self.purge_orphaned_data:
-      tf.logging.warning(
-          '--db_import does not yet support purging orphaned data')
-    self._max_reload_threads = max_reload_threads
-    if self._max_reload_threads > 1:
-      tf.logging.warning(
-          '--db_import does not yet support more than one reload thread')
+    conn.execute('PRAGMA journal_mode=wal')
+    conn.execute('PRAGMA synchronous=normal')  # Recommended for WAL mode
+    sqlite_writer.initialize_schema(conn)
```

Comment:
Just wondering, why do we use python schema generator in all case? Is this to prevent any HTTP request from failing due to no summary write?

Reply:
Yes, it's to avoid the race condition where HTTP requests fail on a "table not found" query failure if they arrive before the background reload has found any event data to import.